File: LeetCode/Easy/0933-number-of-recent-calls/0933-number-of-recent-calls.py
class RecentCounter:
    def __init__(self):
        self.counter = 0
        self.queue = []

    def ping(self, t: int) -> int:
        self.queue.append(t)
        self.counter += 1

        while self.queue[0] < t - 3000:
            self.queue.pop(0)
            self.counter -= 1

        return self.counter

File: LeetCode/Easy/0933-number-of-recent-calls/0933-number-of-recent-calls.rs
struct RecentCounter {
    counter: usize,
    queue: Vec<i32>,
}

impl RecentCounter {
    fn new() -> Self {
        RecentCounter {
            counter: 0,
            queue: Vec::new(),
        }
    }

    fn ping(&mut self, t: i32) -> i32 {
        self.queue.push(t);
        self.counter += 1;

        while self.queue[0] < t - 3000 {
            self.queue.remove(0);
            self.counter -= 1;
        }

        return self.counter as i32;
    }
}
